src/Enzo/enzo_EnzoInitialM1Closure.cpp

```diff
@@ -0,0 +1,105 @@
+// See LICENSE_CELLO file for license and copyright information
+
+/// @file     enzo_EnzoInitialM1Closure.cpp
+/// @author   William Hicks (whicks@ucsd.edu)
+/// @date     Tue September 6 2022
+/// @brief    [\ref Enzo] EnzoMethodM1Closure initializer
+
+#include "cello.hpp"
+#include "enzo.hpp"
+
+//----------------------------------------------------------------------
+
+EnzoInitialM1Closure::EnzoInitialM1Closure
+(const EnzoConfig * enzo_config) throw ()
+  : Initial(enzo_config->initial_cycle, enzo_config->initial_time)
+{
+  return;
+}
+
+//----------------------------------------------------------------------
+
+void EnzoInitialM1Closure::pup (PUP::er &p)
+{
+  // NOTE: update whenever attributes change
+
+  TRACEPUP;
+
+  Initial::pup(p);
+
+}
+
+//----------------------------------------------------------------------
+
+void EnzoInitialM1Closure::enforce_block
+(Block * block, const Hierarchy  * hierarchy) throw()
+{
+   Field field = block->data()->field();
+   int mx,my,mz;  
+   field.dimensions(0,&mx, &my, &mz); //field dimensions, including ghost zones
+  
+   const EnzoConfig * enzo_config = enzo::config();
+         EnzoUnits * enzo_units = enzo::units();
+
+   double inverse_Nunit = 1.0 / enzo_units->photon_number_density();
+   double inverse_Funit = inverse_Nunit * enzo_units->time()/enzo_units->length();
+   double Nmin = enzo_config->method_m1_closure_min_photon_density;
+
+   for (int i=0; i<enzo_config->method_m1_closure_N_groups; i++) {
+    std::string istring = std::to_string(i);
+    enzo_float *  N_i = (enzo_float *) field.values("photon_density_" + istring);
+    enzo_float * Fx_i = (enzo_float *) field.values("flux_x_" + istring);
+    enzo_float * Fy_i = (enzo_float *) field.values("flux_y_" + istring);
+    enzo_float * Fz_i = (enzo_float *) field.values("flux_z_" + istring);
+    for (int j=0; j<mx*my*mz; j++)
+    {
+      N_i [j] = Nmin  * inverse_Nunit;
+      Fx_i[j] = 1e-16 * inverse_Funit;
+      Fy_i[j] = 1e-16 * inverse_Funit;
+      Fz_i[j] = 1e-16 * inverse_Funit;
```

Comment:
Out of curiosity, do problems arise when `"flux_x_i"` is zero? If so, that could be worth noting somewhere.

Reply:
Yes -- this results in NaN's that originate from `EnzoMethodM1Closure::get_reduced_variables()`. I'll add a comment explaining this